src/algebra/pointwise.lean

```diff
@@ -578,6 +578,11 @@ section monoid
   multiplication `*` as "multiplication". -/
 @[derive inhabited] def set_semiring (α : Type*) : Type* := set α
 
+/-- The natural order by inclusion on subsets of `α` is available on `set_semiring α`. -/
+attribute [derive partial_order] set_semiring
+/-- The natural order by inclusion on subsets of `α` has `∅` as its bottom element. -/
+attribute [derive order_bot] set_semiring
```

Comment:
Thank you!  I had placed the doc-strings mostly since I was baffled by the `derive`.  I agree that once you have internalised what is going on, the doc-string is redundant!